File: swift/Sources/allmark/RenderToConsole.swift
import Foundation
import Collections

@MainActor
func renderToConsole(doc: MarkdownNode, renderers: OrderedDictionary<String, Renderer> = consoleRenderers) -> String {
	var state = RendererState(
		renderers: renderers,
		output: "",
		footnotes: [],
		depth: 0,
		quoteDepth: 0
	)

	renderChildren(node: doc, state: &state)

	state.output = state.output.trimmingCharacters(in: .newlines)

	return state.output
}

File: web/src/renderToConsole.ts
import renderChildren from "./render/renderChildren";
import consoleRenderers from "./rulesets/consoleRenderers";
import type ConsoleRendererState from "./types/ConsoleRendererState";
import type MarkdownNode from "./types/MarkdownNode";
import type Renderer from "./types/Renderer";

export default function renderToConsole(
	doc: MarkdownNode,
	renderers?: Map<string, Renderer>,
): string {
	renderers ??= consoleRenderers;

	let state: ConsoleRendererState = {
		renderers: consoleRenderers,
		output: "",
		footnotes: [],
		depth: 0,
		quoteDepth: 0,
	};

	renderChildren(doc, state);

	state.output = state.output.replace(/\n+$/, "");

	return state.output;
}
